cli/functionary/utils.py

```diff
@@ -89,3 +93,150 @@ def format_results(results, title="", excluded_fields=[]):
         table.add_row(*row_data)
         first_row = False
     console.print(table)
+
+
+def _get_package_functions(path):
+    """Returns list of dictionary of functions in the package.yaml"""
+    with open(path, "r") as package_yaml:
+        package_definition = yaml.safe_load(package_yaml)
+    return package_definition["package"]
+
+
+def _get_functions_for_package(package_name):
+    """
+    Takes in package_name, returns functions associated with package
+    """
+    packages = get("packages")
+    functions = get("functions")
+    if not packages or not functions:
+        return {}
+    package_id = _get_package_id(packages, package_name)
+    package_functions = sort_functions_by_package(functions)
+
+    return package_functions[package_id]
+
+
+def _get_package_id(packages, package_name):
+    """
+    Takes in packages, desired package_name, returns associated package id
+    """
+    for package in packages:
+        if package["name"] == package_name:
+            return package["id"]
+
+
+def sort_functions_by_package(functions):
+    """
+    Sorts functions by their package id
+    """
+    functions_lookup = {}
+    package_id = 0
+    for function in functions:
+        package_id = function["package"]
+        if package_id in functions_lookup:
+            functions_lookup[package_id].append(function)
+        else:
+            functions_lookup[package_id] = [function]
+    return functions_lookup
+
+
+def check_changes(path):
+    """
+    Checks for discrepancies between the package.yaml and the API functions, returns
+    whether or not changes occurred
+    """
+    packagefunctions = _get_package_functions(path)["functions"]
+    apifunctions = _get_functions_for_package(_get_package_functions(path)["name"])
+
+    # Nothing in the api
+    if not apifunctions:
+        newfunctions = packagefunctions
+        removedfunctions = {}
+        updatedfunctions = {}
+    else:
+        newfunctions = new_functions(packagefunctions, apifunctions)
+        removedfunctions = removed_functions(packagefunctions, apifunctions)
+        updatedfunctions = updated_functions(packagefunctions, apifunctions)
+
+    if newfunctions:
+        bullet_list(newfunctions, "New functions")
+    if removedfunctions:
+        bullet_list(removedfunctions, "Removed Functions")
+    if updatedfunctions:
+        _format_updated_functions(updatedfunctions, "Updated Functions")
+
+    return newfunctions or removedfunctions or updatedfunctions
```

Comment:
There's a little bit of ambiguity with what you are returning here.  `newfunctions or removedfunctions or updatedfunctions` will return the first of those values that isn't empty.  You then check that value for truthiness later on.  What would be more explicit would be to return a boolean here.

Reply:
fixed